specific/cmo_purchase_level_validation/models/purchase_level_validation.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+from openerp import fields, models, api, _
+from openerp.exceptions import ValidationError
+
+
+class PurchaseLevelValidataion(models.Model):
+    _name = 'purchase.level.validation'
+
+    level = fields.Integer(
+        string='Level',
+        required=True,
+    )
+    limit_amount = fields.Float(
+        string='Limit Amount',
+        required=True,
+    )
+    operating_unit_id = fields.Many2one(
+        'operating.unit',
+        string='Operating Unit',
+    )
+    user_ids = fields.Many2many(
+        'res.users',
+        'res_user_rel', 'validation_id', 'user_id',
+        string='Users',
+    )
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    level_id = fields.Many2one(
+        'purchase.level.validation',
+        string='Level Validation',
+        track_visibility='onchange',
+        copy=False,
+    )
+    approval_ids = fields.Many2many(
+        'res.users',
+        'res_approval_rel', 'purchase_id', 'user_id',
+        string='Approval',
+        track_visibility='onchange',
+        copy=False,
+    )
+    approve_level = fields.Char(
+        string='Approve Level ',
+        compute='_compute_approve_level',
+    )
+
+    @api.depends('level_id', 'approve_level')
+    def _compute_approve_level(self):
+        for order in self:
+            if order.level_id:
+                order.approve_level = str(order.level_id.level)
+
+    @api.multi
+    def action_check_approval(self):
+        amount_untaxed = self.amount_untaxed
+        target_levels = self.env['purchase.level.validation'].search([
+            ('operating_unit_id', 'in',
+                self.env.user.operating_unit_ids.mapped('id')),
+            ('limit_amount', '<=', amount_untaxed),
+        ]).sorted(key=lambda r: r.level)
```

Comment:
I think, you can use `.sorted('level')`
Please check first, I am not very sure.

Reply:
.sorted('level') this's not work.